iroha_p2p/src/network.rs

```diff
@@ -0,0 +1,153 @@
+use std::{collections::HashMap, io::Error};
+
+use async_stream::stream;
+use futures::Stream;
+use iroha_actor::{broker::Broker, Actor, Addr, Context, ContextHandler, Handler};
+use iroha_logger::warn;
+use tokio::net::{TcpListener, TcpStream};
+use ursa::{encryption::symm::Encryptor, kex::KeyExchangeScheme};
+
+use crate::peer::{Peer, State, PeerId};
+use parity_scale_codec::Encode;
+
+/// Main network layer structure, that is holding connections, called [`Peer`]s.
+#[derive(Debug)]
+pub struct Network<T: Encode + Send + Clone + 'static, K: KeyExchangeScheme + Send + 'static, E: Encryptor + Send + 'static> {
+    peers: HashMap<PeerId, Addr<Peer<T, K, E>>>,
+    listener: Option<TcpListener>,
+    broker: Broker,
+    //handler: AlwaysAddr<A>
+}
```

Comment:
There is no need to take Network type of message as a parameter. It should handle sending all messages which implement `Encode` and receive everything which implements `Decode`

Reply:
Same for peer.

Reply:
Guess we can send message to network directly, so no need to subscribe to `Post<T>`

Reply:
Got the problem right now. Guess you need to put `Decode` also there